chore: add integrity tests

joneff opened the pull request:
Fixes: telerik/kendo/issues/7847
Related: telerik/kendo-theme-default/issues/720

@gyoshev  Should we include `karma start` in test or lint npm tasks? Perhaps we need another one for test-watch

gyoshev commented:
We usually have two tasks, one inside the [`npm test` script](https://github.com/telerik/kendo-theme-default/blob/develop/package.json#L39) and one for development, e.g. `npm run watch-test` (like in Angular and React), with `singleRun: false`.

joneff commented:
Point taken. I will update shortly

gyoshev commented on package.json:
```diff
@@ -36,9 +36,14 @@
+    "ts-single": "./node_modules/.bin/tsc --project ./tests/",
+    "ts-watch": "./node_modules/.bin/tsc --watch --project ./tests/",
+    "karma-watch": "karma start --auto-watch --no-single-run",
+    "karma-single": "karma start --no-auto-watch --single-run --browsers PhantomJS"
```
If I understand correctly, tests will run in PhantomJs as part of the theme verification, and with Chrome and PhantomJS locally. It seems that if we use only [Chrome headless](https://developers.google.com/web/updates/2017/06/headless-karma-mocha-chai) this may be simplified, and will require one less runner.

gyoshev commented on package.json:
```diff
@@ -68,16 +73,25 @@
     "sass-lint": "^1.7.0",
     "sassdoc": "^2.1.20",
     "semantic-release": "^6.3.6",
+    "typescript": "^2.6.1",
```
I don't see much justification for using TypeScript in the tests so far, apart from type completion for the `metrics` object. Perhaps we can shave off some of the dependencies and configuration boilerplate if we use JS?

gyoshev commented on tests/button-qunit.ts:
```diff
@@ -0,0 +1,16 @@
+        TestHelper.loadFixture( "/base/tests/fixtures/button.html" );
+    },
+    after: function() {
+        //kendo.destroy(QUnit.fixture);
```
Can be safely deleted.

gyoshev commented on .config/karma.conf.js:
```diff
@@ -0,0 +1,84 @@
+            "dist/all.css",
+            "tests/fixtures/**/*.html",
+            "tests/data/metrics.js",
+            "tests/**/*-qunit.js"
```
I'm assuming that in the different themes, the tests and fixtures will be reused with a different metrics object, is that correct? If so, what will be the configuration for that?

gyoshev commented on tests/lib/helpers.ts:
```diff
@@ -0,0 +1,101 @@
+    QUnit: QUnit
+}
+
+class TestHelper {
```
It appears that the [Karma html2js preprocessor](https://github.com/karma-runner/karma-html2js-preprocessor) allows fixtures to be loaded directly by Karma. Does it fit the use-case?

joneff commented:
@gyoshev i've removed the TS dep and cleaned up the code. Also, I switched to ChromeHeadless per suggestion.

On html2js, the simple fixture loading is working, thus far. Should it fail to meet any of our cases, we'll switch to and alternative.

On multiple theme setup: yes, we are going to have multiple config files and we are going to feed the file via a param. However, I'd like to see how we go with the theme repo restructure first.
